src/behaviors/paginatable-actions.spec.js

```diff
@@ -0,0 +1,74 @@
+/*eslint-env node, jasmine*//*global module inject*/
+/*eslint-disable max-statements, max-params*/
+import angular from 'angular';
+import 'angular-mocks';
+import 'luxyflux/ng-luxyflux';
+
+import {
+  Annotations,
+  Actions,
+  PaginatableActions,
+  PaginatableActionsBehavior
+} from 'anglue/anglue';
+
+describe('PaginatableActions', () => {
+  // Clear the AnnotationCache for unit tests to ensure we create new annotations for each class.
+  beforeEach(() => {
+    Annotations.clear();
+  });
+
+  describe('PaginatableActionsBehavior', () => {
+    let mockInstance, behavior;
+
+    beforeEach(() => {
+      behavior = new PaginatableActionsBehavior(mockInstance);
+    });
+
+    it('should have a changeLimit function', () => {
+      expect(behavior.changeLimit).toBeDefined();
+    });
+
+    it('should have a changePage function', () => {
+      expect(behavior.changePage).toBeDefined();
+    });
+  });
+
+  describe('@PaginatableActions() decorator', () => {
+    @Actions()
+    @PaginatableActions()
+    class PaginatableTestActions {}
+
+    let testActions;
+    let appDispatcher;
+
+    class MockApplicationDispatcher {
+      dispatch() {}
+    }
+
+    angular
+      .module('actionsApp', [
+        'luxyflux',
+        PaginatableTestActions.annotation.module.name
+      ])
+      .service('ApplicationDispatcher', [
+        function () {
+          return new MockApplicationDispatcher();
+        }
+      ]);
+
+    beforeEach(module('actionsApp'));
+    beforeEach(inject((_PaginatableTestActions_, _ApplicationDispatcher_) => {
+      testActions = _PaginatableTestActions_;
+      appDispatcher = _ApplicationDispatcher_;
+
+      spyOn(appDispatcher, 'dispatch');
+    }));
+
+
+    it('should dispatch an action', () => {
+      expect(testActions.changeLimit(1));
```

Comment:
So this breaks with: 'undefined is not an object (evaluating 'this.dispatch.apply)

Logical because the behavior doesn't have the dispatcher available. Any idea on how we can make this work?